pgactivity/views.py

```diff
@@ -184,12 +169,57 @@ def header(
     ui: UI,
     *,
     host: Host,
-    dbinfo: DBInfo,
     pg_version: str,
-    tps: int,
-    active_connections: int,
+    server_informations: ServerInformations,
     system_info: Optional[SystemInfo] = None,
+    header_visibility_info: HeaderVisibilityInfo,
 ) -> Iterator[str]:
+    @functools.singledispatch
+    def render(x: NoReturn) -> str:
+        if x is None:
+            return "-"
+        raise AssertionError(f"not implemented for type '{type(x).__name__}'")
+
+    @render.register(str)
+    def render_str(s: str) -> str:
+        return term.bold_green(s)
+
+    @render.register(int)
+    def render_int(n: int) -> str:
+        return term.bold_green(str(n))
+
+    @render.register(Pct)
```

Comment:
This doesn't work. My `Pct` objects are seen as `floats`.
Is the approach flawed ? Is there a way to "copy" the class instead of inheriting it ?

Reply:
Seems to work on my side with the following example code:
```python
from functools import singledispatch                                
from typing import Any               
                                     
class pct(float):                    
    pass                             
                                     
@singledispatch                      
def fun(arg: Any) -> str:            
    return "any"                     
                                     
@fun.register(pct)                   
def fun_pct(arg: pct) -> str:        
    return "pct"                     
                                     
@fun.register(float)                 
def fun_float(arg: float) -> str:    
    return "float"                   
                                     
@fun.register(int)                   
def fun_int(arg: int) -> str:        
    return "int"                     
                                     
assert fun.dispatch(float) is fun_float
assert fun.dispatch(pct) is fun_pct
assert fun.dispatch(int) is fun_int

assert fun(pct(1.2)) == 'pct'
assert fun(0.1) == 'float'
```

But I see below that `render_pct()` and `render_float()` have the same implementation...

Reply:
One has a "%" the other has not.

Reply:
didn't pay attention to the assert with the functions thats neat ...

Reply:
> One has a "%" the other has not.

Indeed ;)

So does it work?

Reply:
Nope. Will look into why it's not working asap.

Reply:
well, that's lovely. 
I changed it from my `object.__setattr__` to the `property` and it works.